src/components/message/message.spec.js

```diff
@@ -0,0 +1,144 @@
+import React from 'react';
+import TestRenderer from 'react-test-renderer';
+import 'jest-styled-components';
+import { shallow } from 'enzyme';
+import OptionsHelper from '../../utils/helpers/options-helper/options-helper';
+import MessageStyle from './message.style';
+import Message from './message.component';
+import classicTheme from '../../style/themes/classic';
+import CloseIcon from './close-icon/close-icon.component';
+import { assertStyleMatch } from '../../__spec_helper__/test-utils';
+
+function render(props, renderer = TestRenderer.create) {
+  return renderer(<MessageStyle { ...props }>Message</MessageStyle>);
+}
+
+describe('Message', () => {
+  describe('when rendered', () => {
+    it('should match the snapshot', () => {
+      const wrapper = shallow(<Message>Message</Message>);
+      expect(wrapper).toMatchSnapshot();
+    });
+  });
+
+  describe('when onDismiss function is not provided', () => {
+    it('not renders the close icon when the onDismiss prop ', () => {
+      const wrapper = shallow(<Message>Message</Message>);
+      const closeIcon = wrapper.find(CloseIcon);
+      expect(closeIcon.length).toEqual(0);
+    });
+  });
+
+  describe('when onDismiss function is provided', () => {
```

Comment:
I think we can have here just one test about dissmiss and also instead of create everytime `shallow(<Message>Message</Message>)` we can create it in upper scope like
```js
let wrapper;
```
then use `beforeEach` to create the behavior for every test in the scope
```
beforeEach(() => {wrapper = shallow(<Message>Message</Message>)})
```
and if u need to set props then u can use `setProps` function
```js
wrapper.setProps({onDismiss})
```
Example:
```js
describe('Message', () => {
let wrapper;

beforeEach(()=>{
  wrapper = shallow(<Message>Message</Message>);
})

describe('when rendered', () => {
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe('when onDismiss function is not provided', () => {
  it('not renders the close icon when the onDismiss prop ', () => {
    const closeIcon = wrapper.find(CloseIcon);
      expect(closeIcon.length).toEqual(0);
    });
  });

describe('when onDismiss function is provided', () => {
it('should render the close icon', () => {
      const onDismiss = jest.fn();
      wrapper.setProps({onDissmiss})
      const closeIcon = wrapper.find(CloseIcon);
      expect(closeIcon.length).toEqual(1);
    });
  });
```